radixdlt/src/main/java/org/radix/network/messages/PeersMessage.java

```diff
@@ -15,48 +15,62 @@
  * language governing permissions and limitations under the License.
  */
 
-package org.radix.network.messages;
-
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-
+package org.radix.network.messages;
+
+import java.util.List;
+import java.util.Objects;
+
+import com.google.common.collect.ImmutableList;
 import org.radix.network.messaging.Message;
 
 import com.radixdlt.network.addressbook.Peer;
-import com.radixdlt.serialization.DsonOutput;
-import com.radixdlt.serialization.DsonOutput.Output;
-import com.radixdlt.serialization.SerializerId2;
-
-import com.fasterxml.jackson.annotation.JsonProperty;
-
-@SerializerId2("network.message.peers")
-public final class PeersMessage extends Message
-{
-	@JsonProperty("peers")
-	@DsonOutput(Output.ALL)
-	private List<Peer> peers = new ArrayList<>();
-
-	private PeersMessage() {
-		super(0);
-		// For serializer
-	}
-
+import com.radixdlt.serialization.DsonOutput;
+import com.radixdlt.serialization.DsonOutput.Output;
+import com.radixdlt.serialization.SerializerId2;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@SerializerId2("network.message.peers")
+public final class PeersMessage extends Message
+{
+	@JsonProperty("peers")
+	@DsonOutput(Output.ALL)
+	private ImmutableList<Peer> peers = ImmutableList.of();
```

Comment:
I think there is a bug here.  In PeerManager.java, circa line 267, we have the code:
```
peersMessage.getPeers().add(p);
```

This change clearly breaks that quite badly.  Can we modify the code in PeersManager to construct with an ImmutableList once the list is built, and get rid of setPeers(...) altogether?

Something like this:
```
// Constructor
public PeersMessage(int magic, Collection<Peer> peers) {
    super(magic);
    this.peers = ImmutableList.copyOf(peers);
}
```

Code in PeersManager
```
// Deliver known Peers in its entirety, filtered on whitelist and activity
// Chunk the sending of Peers so that UDP can handle it
List<Peer> peersList = Lists.newArrayList();
List<Peer> peers = addressbook.peers()
	.filter(Peer::hasNID)
	.filter(StandardFilters.standardFilter(localSystem.getNID(), whitelist))
	.filter(StandardFilters.recentlyActive(this.recencyThreshold))
	.collect(Collectors.toList());

for (Peer p : peers) {
	if (p.getNID().equals(peer.getNID())) {
		// Know thyself
		continue;
	}

	peersList.add(p);
	if (peersList.size() == peerMessageBatchSize) {
		messageCentral.send(peer, new PeersMessage(this.universeMagic, peersList));
		peersList.clear();
	}
}

if (!peersList.isEmpty()) {
	messageCentral.send(peer, new PeersMessage(this.universeMagic, peersList));
}
```

Reply:
a similar change in PeerManager was already included in this PR: https://github.com/radixdlt/radixdlt-core/pull/230/files#diff-ff41b2829e41459497a0d7c04df7fe50R268

I'll change it to what you suggested though